src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -313,9 +336,11 @@ private float addRow(
           RowBufferStats.getCombinedStats(statsMap.get(columnName), forkedColStats.getValue()));
     }
 
-    // Increment null count for column missing in the input map
+    // Increment null count for column and its sub-columns missing in the input map
     for (String columnName : Sets.difference(this.fieldIndex.keySet(), inputColumnNames)) {
-      statsMap.get(columnName).incCurrentNullCount();
+      statsFinder
+          .getSubColumns(columnName)
+          .forEach(subColumn -> statsMap.get(subColumn).incCurrentNullCount());
```

Comment:
nullref if statsMap.get(subColumn) returns null.
can that happen?

Reply:
No as both `subColumnFinder` and `statsMap` are built from the same schema. Added check here in case.